OSDOCS-2955: Adds common terms section to the Architecture Book (#49007)
openshift/openshift-docs: modules/openshift-architecture-common-terms.adoc

// Module included in the following assemblies:
//
// * architecture/index.adoc

:_content-type: REFERENCE
[id="openshift-architecture-common-terms_{context}"]
= Common terms

This glossary defines common terms that are used in the architecture content. These terms help you understand {product-title} architecture effectively.

access policies::

> A set of roles that dictate how users, applications, and entities within a cluster interacts with one another. Access policies increase cluster security.

Reviewer: I think you should use an article before access policies here because access is a verb but its being used as a noun in this sentence. Maybe: An access policy increases cluster security.

Author: Addressed

> Admission plug-ins::

Reviewer 2: Not sure if we have terms defined elsewhere in the doc set, but just a general suggestion for all of these would be to lowercase terms (i.e. show the correct capitalization). 

Like:

```
access policies:: ...

admission plug-ins:: ...
```

Looking at these, users might think that it should be written as capitalized "Admission plug-ins" even in the middle of a sentence. 

The IBM style guide and SSG both use lowercase/proper capitalization when listing their words. And the IBM style guide gives this guidance: 

> Do not capitalize a term in the glossary unless it is a proper noun.

Author: acked

Author: @bergerhoffer Does the capitalization look correct now? I made some changes to the terms. Seems like we have a merged instance of glossary items here: https://docs.openshift.com/container-platform/4.11/installing/index.html#install-openshift-common-terms_ocp-installation-overview

Admission plug-ins enforce security policies, resource limitations, or configuration requirements.

authentication::
To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.

bootstrap::
A temporary machine that runs minimal Kubernetes and deploys the {product-title} control plane.

certificate signing requests (CSRs)::
A resource requests a denoted signer to sign a certificate. This request might get approved or denied.

> cluster Version Operator (CVO)::

Reviewer 2: Cluster here should have stayed capitalized

Author: acked

An Operator that checks with the {product-title} Update Service to see the valid updates and update paths based on current component versions and information in the graph.

Compute nodes::
Nodes that are responsible for executing workloads for cluster users. Also known as worker nodes.

Configuration drift::
A situation where the configuration on a node does not match what the machine config specifies.

Containers::

> Lightweight and executable images that consist software and all its dependencies. Containers virtualize the operating system, as a result, you can run containers anywhere, from a data centre to a public or private cloud to your local host.

Author: Addressed

Container orchestration engine::
Software that automates the deployment, management, scaling, and networking of containers.

Container workloads::
Applications that are packaged and deployed in containers.

Control groups (cgroups)::

> cgroups Partition sets of processes into groups to manage and limit the resources processes consume.

Author: Addressed

control plane::

> A container orchestration layer that exposes the API and interfaces to define, deploy, and manage the life cycle of containers. Also known as control plane machines.

Reviewer: ```suggestion
A container orchestration layer that exposes the API and interfaces to define, deploy, and manage the life cycle of containers. Control planes are also known as control plane machines.
```

Author: Addressed

CRI-O::
A Kubernetes native container runtime implementation that integrates with the operating system to deliver an efficient Kubernetes experience.

Deployment::

> A Kubernetes resource object that maintains the life cycle of an application.

Author: Addressed

> dockerfile::

Reviewer 2: Dockerfile should stay capitalized

Author: Addressed

A text file that contains the user commands to perform on a terminal to assemble the image.

hosted control planes::
A {product-title} feature that enables hosting a control plane on the {product-title} cluster from its data plane and workers. This model performs following actions:

* Optimize infrastructure costs required for the control planes.
* Improve the cluster creation time.
* Enable hosting the control plane using the Kubernetes native high level primitives. For example, deployments, stateful sets.
* Allow a strong network segmentation between the control plane and workloads.

hybrid cloud deployments::
Deployments that deliver a consistent platform across bare metal, virtual, private, and public cloud environments. This offers speed, agility, and portability.

> ignition::

Reviewer 2: Ignition should stay capitalized

Author: Addressed

A utility that {op-system} uses to manipulate disks during initial configuration. It completes common disk tasks, including partitioning disks, formatting partitions, writing files, and configuring users.

Installer-provisioned infrastructure::
The installation program deploys and configures the infrastructure that the cluster runs on.

Kubelet::

> A **primary node** agent that runs on each node in the cluster to ensure that containers are running in a pod.

Author: Addressed- removed the bold and added primary node to the list of common terms

kubernetes manifest::
Specifications of a Kubernetes API object in a JSON or YAML format. A configuration file can include deployments, config maps, secrets, daemon sets.

> machine Config Daemon (MCD)::

Reviewer 2: For MCD and MCO the "Machine" should stay capitalized

Author: Addressed

Machine config pools (MCP)::
A group of machines, such as control plane components or user workloads, that are based on the resources that they handle.

Metadata::
Additional information about cluster deployment artifacts.

Microservices::
An approach to writing software. Applications can be separated into the smallest components, independent from each other by using microservices.

Mirror registry::
A registry that holds the mirror of {product-title} images.

Monolithic applications::
Applications that are self-contained, built, and packaged as a single piece.

Namespaces::
A namespace isolates specific system resources that are visible to all processes. Inside a namespace, only processes that are members of that namespace can see those resources.

Networking::
Network information of {product-title} cluster.

Node::
A worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine.

{product-title} Update Service (OSUS)::
For clusters with internet access, {op-system-base-full} provides over-the-air updates by using an {product-title} update service as a hosted service located behind public APIs.

> {product-title} CLI (`oc`)::

Author: acked

A command line tool to run {product-title} commands on the terminal.

> {product-title} Dedicated::

Author: acked

A managed {op-system-base} {product-title} offering on Amazon Web Services (AWS) and Google Cloud Platform (GCP). OpenShift Dedicated focuses on building and scaling applications.

{product-title} registry::
A registry provided by {product-title} to manage images.

Operator::
The preferred method of packaging, deploying, and managing a Kubernetes application in an {product-title} cluster. An Operator takes human operational knowledge and encodes it into software that is packaged and shared with customers.

OperatorHub::
A platform that contains various {product-title} Operators to install.

Operator Lifecycle Manager (OLM)::
OLM helps you to install, update, and manage the lifecycle of Kubernetes native applications. OLM is an open source toolkit designed to manage Operators in an effective, automated, and scalable way.

> Over-the-air (OTA) updates::

Reviewer 2: Over should be lowercase

Author: Addresses

The {product-title} Update Service (OSUS) provides over-the-air updates to {product-title}, including {op-system-first}.

pod::
One or more containers with shared resources, such as volume and IP addresses, running in your {product-title} cluster.
A pod is the smallest compute unit defined, deployed, and managed.

private registry::
{product-title} can use any server implementing the container image registry API as a source of the image which allows the developers to push and pull their private container images.

public registry::
{product-title} can use any server implementing the container image registry API as a source of the the image which allows the developers to push and pull their public container images.

{op-system-base} {product-title} Cluster Manager::
A managed service where you can install, modify, operate, and upgrade your {product-title} clusters.

{op-system-base} Quay Container Registry::
A Quay.io container registry that serves most of the container images and Operators to {product-title} clusters.

replication controllers::
An asset that indicates how many pod replicas are required to run at a time.

> Role-based access control (RBAC)::

Reviewer 2: Role should be lowercase

Author: Addressed

A key security control to ensure that cluster users and workloads have only access to resources required to execute their roles.

route::

> Exposes a service to allow for network access to pods from users and applications outside the {product-title} instance.

Reviewer: ```suggestion
Routes expose a service to allow for network access to pods from users and applications outside the {product-title} instance.
```

Author: Addressed

scaling::

> Increasing or decreasing the resource capacity.

Reviewer: ```suggestion
The increasing or decreasing of resource capacity.
```

Author: Addressed

service::

> Exposes a running application on a set of pods.

Reviewer: ```suggestion
A service exposes a running application on a set of pods.
```

Author: Addressed

Source-to-Image (S2I) image::
An image created based on the programming language of the application source code in {product-title} to deploy applications.

storage::
{product-title} supports many types of storage, both for on-premise and cloud providers. You can manage container storage for persistent and non-persistent data in an {product-title} cluster.

> telemetry::

Reviewer 2: I'm pretty sure that Telemetry is capitalized (per how it's used here: https://docs.openshift.com/container-platform/4.11/support/remote_health_monitoring/about-remote-health-monitoring.html)

Author: addressed

Operator Lifecycle Manager (OLM)::
Operator Lifecycle Manager (OLM) helps users install, update, and manage the lifecycle of Kubernetes native applications (Operators) and their associated services running across their OpenShift Container Platform clusters. It is part of the Operator Framework, an open source toolkit designed to manage Operators in an effective, automated, and scalable way.

Red Hat Quay Container Registry::
Is a Quay.io container registry that serves most of the container images and Operators to {product-title } clusters.

Log aggregation::
The logging subsystem aggregates all the logs from your OpenShift Container Platform cluster, such as node system audit logs, application container logs, and infrastructure logs, into a dedicated log store. You can then query, discover, and visualize your log data through the xref:../logging/cluster-logging-visualizer.adoc#cluster-logging-visualizer-using[Kibana interface]

Web console::
A user interface (UI) to manage {product-title}.

OpenShift CLI (`oc`)::
A command line tool to run {product-title} commands on the terminal.

Telemetry::

> A component to collect information such as size, heakth and status of {product-title}.

Reviewer 3: ```suggestion
A component to collect information such as size, health, and status of {product-title}.
```

Author: Acked

* Improves cluster creation time.
* Enable hosting the CP using Kubernetes-native high-level primitives (e.g., deployments, stateful sets, etc).
* Allows strong network segmentation between the control-plane and workloads.

Private registry::
OpenShift Container Platform can utilize any server implementing the container image registry API as a source of image which allows the developers to push and pull their private container images.

Public registry::
OpenShift Container Platform can utilize any server implementing the container image registry API as a source of image which allows the developers to push and pull their public container images.

OpenShift registry::
A registry provided by {product-title} to manage images.

Source-to-Image (S2I) image::
Is an image created based on the programming language of the application source code in {product-title} to deploy applications.

Dockerfile::
A text file that contains the user commands to be performed on a terminal to assemble the image.

Route::
Exposes a service to allow for network access to pods from users and applications outside the OpenShift instance.

Deployment::

> A resource object that maintains the life cycle of an application.

Reviewer 3: ```suggestion
A Kubernetes resource object that maintains the life cycle of an application.
```
More info: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

Author: acked
